### PR TITLE
Merge Django 4.1 branch (at last)

### PR DESCRIPTION
**Pull request by sfu-peter**

### CODE REVIEW

**Review by calvinhp — CHANGES_REQUESTED** [review]
Can't wait to use it!

File: {{cookiecutter.project_slug}}/Makefile [hunk 1]
```diff
@@ -1,48 +1,130 @@
...
+		 python -m piptools compile --upgrade --allow-unsafe --output-file /local/local.txt /local/local.in"
+
+load-dump: destroy-data ## Fresh load the data file passed in
+	@sleep 5
+	pg_restore -h localhost -p $(PGPORT) -j 4 -n public --no-owner -U {{ cookiecutter.project_slug }} -W -d {{ cookiecutter.project_slug }} $(DUMPFILE)
+
+clean-dev: ## Remove the latest build
+	@docker rmi -f {{ cookiecutter.project_slug }}_local_django:dev
+
+up:
```
[the remark this answers is not in the document]
Comment (sfu-peter) [reply {{cookiecutter.project_slug}}/Makefile:111]: The node bits are commented out in the docker-compose anyway?  But I get what you're aiming at.

File: {{cookiecutter.project_slug}}/docker-compose.yml [hunk 2]
```diff
@@ -1,32 +1,31 @@
...
       - local/environment
   django:
-    build:
-      context: .
-      args:
-        DEVEL: 'yes'
     image: '{{ cookiecutter.project_slug }}_local_django:dev'
-    command: 'python manage.py runserver_plus 0.0.0.0:8000'
+    #command: 'python manage.py runserver 0.0.0.0:8000'
+    command: 'gunicorn config.asgi:application -k uvicorn.workers.UvicornWorker'
```
[the remark this answers is not in the document]
Comment (sfu-peter) [reply {{cookiecutter.project_slug}}/docker-compose.yml:16]: The gunicorn command replaces daphne, which is usually a part of channels, which is not supported in this version of the cookiecutter.  The gunicorn/uvicorn command enables django to work in asgi mode without channels, and it's better supported than daphne as well.  But it's true the runserver command should be the uncommented one.  I don't believe runserver_plus can use asgi though?

File: README.md [hunk 3]
```diff
@@ -6,3 +6,22 @@ and Daniel Roy Greenfeld. It also borrows heavily from the ideas in the
 PyPA [warehouse](https://github.com/pypa/warehouse) as I feel they have 
 made an amazing developer onboarding experience for a very complicated
 project.
+
+Creating a new project using this repo:
```
[the remark this answers is not in the document]
Comment (sfu-peter) [reply README.md:10]: I don't think this is necessary--for most people, they only use the cookiecutter very rarely (maybe only once?)  Directing them to install the whole pipx infrastructure just to generate a one-off project seems invasive and heavy-handed. The in-project venv is a workable solution that can easily be removed when they've generated their project into another directory...?

[the remark this answers is not in the document]
Comment (calvinhp) [reply README.md:10]: Adding to the issues list, I'd love to say we are highly opinionated and want the world to discover the join that is `pipx`!

**Pull request merged**